lib/widgets/update_expense.dart

```diff
@@ -1,98 +1,158 @@
-          child: ListBody(children: [
-            TextFormField(
-              textAlign: TextAlign.center,
-              controller: _editExpenseValueController,
-              style: const TextStyle(fontSize: 70, color: Colors.black),
-              autofocus: true,
-              autocorrect: false,
-              decoration: const InputDecoration(hintText: "0.00"),
-              textInputAction: TextInputAction.next,
-              keyboardType: TextInputType.number,
-            ),
-            const SizedBox(
-              height: 20.0,
-            ),
-            TextFormField(
-              controller: _editExpenseNameController,
-              autofocus: true,
-              autocorrect: false,
-              decoration: const InputDecoration(hintText: "Expense Name"),
-              textInputAction: TextInputAction.done,
-            ),
-            const SizedBox(
-              height: 20.0,
-            ),
-            FutureBuilder<List<ExpenseCategory?>?>(
-                future: apiService.getExpenseCategories(),
-                builder: (context, snapshot) {
-                  return DropdownButtonFormField<ExpenseCategory>(
-                    hint: Text(expenseItem.expensecategory.categoryname),
-                    onChanged: (newValue) {
-                      _selectedExpenseCategory = newValue!;
-                    },
-                    // value: selectedCategory,
-                    items: snapshot.data
-                        ?.map((ec) => DropdownMenuItem<ExpenseCategory>(
-                              value: ec,
-                              child: Text(ec!.categoryname),
-                            ))
-                        .toList(),
-                  );
-                }),
-            const SizedBox(
-              height: 20.0,
-            ),
-          ]),
+      content: Form(
+        key: formGlobalKey,
+        child: Container(
+          padding: const EdgeInsets.all(16.0),
+          width: double.infinity,
+          child: SingleChildScrollView(
+            child: ListBody(children: [
+              TextFormField(
+                textAlign: TextAlign.center,
+                controller: _editExpenseValueController,
+                style: const TextStyle(fontSize: 70, color: Colors.black),
+                autofocus: true,
+                autocorrect: false,
+                decoration: const InputDecoration(hintText: "0.00"),
+                textInputAction: TextInputAction.next,
+                keyboardType: TextInputType.number,
+                inputFormatters: <TextInputFormatter>[
+                  FilteringTextInputFormatter.digitsOnly
+                ],
+                validator: (value) {
+                  const _validationError = 'Enter a valid expense value';
+                  if (value == null || value.isEmpty) {
+                    return _validationError;
+                  }
+                  if (double.parse(value) <= 0) {
```

Comment:
Use validator from above

```suggestion
                  final parsed = double.tryParse(value);
                  if (parsed == null || parsed <= 0) {
```